### ISSUE TITLE
"no-unused-vars" rule should not fail when vars are actually used

### ISSUE THREAD

**Issue by tjwebb** [eslint/eslint#6243]
**What version of ESLint are you using?**
2.10

**What parser (default, Babel-ESLint, etc.) are you using?**
none

**Please show your full configuration:**
```
    'no-unused-vars': [2, {'vars': 'local', 'args': 'none'}],
```

**What did you do? Please include the actual source code causing the issue.**
```js
let ParcelService
module.exports = class ParcelController extends Controller {
  constructor (app) {
    super(app)
    ParcelService = app.services.ParcelService
  }
}
```

**What did you expect to happen?**
I expected there to be no rule violation

**What actually happened? Please include the actual, raw output from ESLint.**
'ParcelService' is defined but never used (no-unused-vars)

**Comment by ilyavolodin** [comment]
Why do you think `ParcelService` is used? It's assigned to, but nothing in your code example uses the value of that variable. Based purely on the code that you posted, you can safely remove that variable without affecting anything.

**Comment by nzakas** [comment]
Per the [docs](http://eslint.org/docs/rules/no-unused-vars#rule-details), a variable is not considered to be used unless it's value is read. In this example, the variable is only ever assigned to but never read, so it is unused.
